storage/kubernetes/storage.go

```diff
@@ -35,9 +35,12 @@ const (
 )
 
 // Config values for the Kubernetes storage type.
+//
+// TODO(rithujohn): make useCRD = True the default behavior.
 type Config struct {
 	InCluster      bool   `json:"inCluster"`
 	KubeConfigFile string `json:"kubeConfigFile"`
+	UseCRD         bool   `json:"useCRD"` // Flag option to use CRDs instead of TPRs
```

Comment:
I know this sounds kind of silly, but I would make the flag `UseTPR` instead of `UseCRD`. You can bump to a new major version for this, so clients have to explicitly set `UseTPR` if they want to. Then removing the flag later will be much cleaner and require no changes from conforming clients.

(Note that this applies to all the boolean vars you plumb through everywhere).